### ISSUE TITLE
[Bug]: babel-plugin-transform-modules-umd using in babel standablong. the generated code error while transform the same dependency name.

### ISSUE THREAD

**Issue by XiaoWinter** [babel/babel#15516]
### 💻

- [X] Would you like to work on a fix?

### How are you using Babel?

babel-loader (webpack)

### Input code

```js
import { openMacroMessage, booleanConfirm } from "tms-front/lib/common/utils";

import { getCurrentOrganizationId } from "utils/utils";

console.log("openMacroMessage", openMacroMessage);

console.log("booleanConfirm", booleanConfirm);

console.log("getCurrentOrganizationId", getCurrentOrganizationId);
```
I found a bug while using the babel-plugin-transform-modules-umd plugin, In the source code above, there are imports for both "tms-front/lib/common/utils" and "utils/utils", but in the resulting code, they have the same name at line 10, causing an error. 

I have noticed that the plugin just splits the last word when translating dependencies, as demonstrated in my example.

 You can reproduce this bug on your official website， I have posted a picture of the error below.
 
![image](https://user-images.githubusercontent.com/41815901/227697869-9ac91a03-1f9b-4852-874d-7dc6e07294c1.png)

### Current and expected behavior

I suggest that the same dependency name in the resulting code will be added number suffix in sequence.

![无标题1](https://user-images.githubusercontent.com/41815901/227699574-b194ceb8-e8ae-48c8-bc34-80691337891e.png)

### Environment

-

**Comment by XiaoWinter** [comment]
> Hey @XiaoWinter! We really appreciate you taking the time to report an issue. The collaborators on this project attempt to help as many people as possible, but we're a limited number of volunteers, so it's possible this won't be addressed swiftly.
> 
> If you need any help, or just have general Babel or JavaScript questions, we have a vibrant [Slack community](https://babeljs.slack.com) that typically always has someone willing to help. You can sign-up [here](https://slack.babeljs.io/) for an invite.

I don't how to fix the issuse, can you give me some document or ponit the dorction to resolve it

**Comment by JLHwung** [comment]
related: https://github.com/babel/babel/issues/4381

**Comment by XiaoWinter** [comment]
> [Playground link](https://babeljs.io/repl#?browsers=defaults&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=JYWwDg9gTgLgBAbzhMBTAdgWQIYGMoSaoDOx2A5qgDRwBGEEANqtugMIToBmwUIcAXzhcC_AEQwQxALQjOMAPSNgtBbgggQnBQFcYwRsTEBuAFCnQkWIjiUYbHVCgYYAeSjlWwAF7Z9nAEkAE0FhUTgxPQNiXX1DE3N1dGImVAA6RghyAAoxFAwcfEISMkoxGnysPAIiUgpUAEozUySU5gys3PpU1g5uXhByugZmXs4ePibEzjb0zJyxOwcnF3dPdB8_YECgoaXHZ3Q3Dy9ff3RgpqA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-3%2Ctypescript&prettier=false&targets=&version=7.22.5&externalPlugins=%40babel%2Fplugin-transform-modules-umd%407.18.6&assumptions=%7B%7D) to repro

Thank you for your attention. I noticed there is a similar issue found by [JLHwung](https://github.com/JLHwung). Also, thank you to [JLHwung](https://github.com/JLHwung).
related: https://github.com/babel/babel/issues/4381

**Comment by nicolo-ribaudo** [comment]
With the proposed fix, changing the order of imports would change the name of the globals they are read from. If two libraries have the same name, there is no good automatic way to disambiguate.

I think the only "fix" I would be comfortable with is adding an option that takes a `(importerFile: stiring, importSpecifier: string) => string` to let users compute the global names the way it works for their libraries.

**Comment by XiaoWinter** [comment]
yes, My idea is not thoughtful. I dno't know others scene, in my project, there are two steps about use this plugin,first I write a js file to defined which module will be mounted 'window'
``` js
function initModule() {
  window.react = require('react');
  window.lodash = require('lodash');
  window.utils = require('a/utils'); 
//  some teammate know babel-plugin-transform-modules-umd don't division ‘a/utils’ and 'b/utils'，they won't import 'b/utils' 
// in tsx file which will be converted by babel/standalong,  instead of use window.utils2 directly
  window.utils2 = require('b/utils'); 
}
```
second, the prepared tsx file will be converted by babel/standalong , the plugin will be run in the duration. then the result js file will be inserted page and run.

in the whole process, the mounted action and coverted action is unralated, users mounted modules by hand.the plugin's action is normal also. Typically, developers don't need to concern themselves with the mechanism. but sometime  Developers must be aware of the mechanism of this plugin and handle module mounting themselves. so I think add a option as a hook is useful ，wihch give developers a way to control the result.  users are able to do something  when the  name of importFile is generated. in my case. if this plugin has a hook,I can  deal with 'b/utils' exactly. to ensure consistency in module names  mounting 
 to window and coverted by this plugin.

``` js
    const res = Babel.transform(source, {
      presets: [
        'react',
        'es2015',
        'stage-3',
        ['typescript', { allExtensions: true, allowNamespaces: true, isTSX: true }],
      ],
      plugins: [
        ['transform-modules-umd',
         {
            afterFileNameGenarate:(generatedFilename,moduleName)=>{
              if(moduleName === 'b/utils') return 'utils2'
          }
         }
       ],
        ['transform-typescript'],
      ],
    });


```
